commcare/src/test/java/org/motechproject/commcare/service/impl/CommcareReportServiceImplTest.java

```diff
@@ -40,17 +51,35 @@ public void setUp() {
 
     @Test
     public void shouldGetReports() {
-        when(commcareHttpClient.reportsListMetadataRequest(configService.getByName(null).getAccountConfig())).thenReturn(getResponseForReportsList());
+        when(commcareHttpClient.reportsListMetadataRequest(configService.getByName(null).getAccountConfig())).thenReturn(getResponseForReportsListMetadata());
 
         ReportsMetadataInfo reportsMetadataInfo = reportService.getReportsList();
 
         assertThat(reportsMetadataInfo.getMetadataInfo().getTotalCount(), equalTo(2));
         assertThat(reportsMetadataInfo.getReportMetadataInfoList().size(), equalTo(2));
         assertThat(reportsMetadataInfo.getReportMetadataInfoList().get(0).getColumns().size(), equalTo(3));
         assertThat(reportsMetadataInfo.getReportMetadataInfoList().get(0).getFilters().size(), equalTo(2));
+        assertThat(reportsMetadataInfo.getReportMetadataInfoList().get(0).getId(), equalTo("9aab0eeb88555a7b3bc8676883e7379a"));
+        assertThat(reportsMetadataInfo.getReportMetadataInfoList().get(0).getTitle(), equalTo("Test Report 1"));
+        assertThat(reportsMetadataInfo.getReportMetadataInfoList().get(0).getColumns().get(0).getId(), equalTo("name"));
+        assertThat(reportsMetadataInfo.getReportMetadataInfoList().get(0).getColumns().get(0).getDisplay(), equalTo("Name"));
+        assertThat(reportsMetadataInfo.getReportMetadataInfoList().get(0).getColumns().get(0).getType(), equalTo(ColumnType.FIELD));
+        assertThat(reportsMetadataInfo.getReportMetadataInfoList().get(0).getFilters().get(0).getType(), equalTo(FilterType.CHOICE_LIST));
+        assertThat(reportsMetadataInfo.getReportMetadataInfoList().get(0).getFilters().get(0).getDatatype(), equalTo(FilterDataType.STRING));
+        assertThat(reportsMetadataInfo.getReportMetadataInfoList().get(0).getFilters().get(0).getSlug(), equalTo("closed"));
```

Comment:
it can be done, using one assert. Remember to overrride equals and hashcode methods in ReportMetadataInfo class